src/installer/BeatPackageCompiler/AgentCustomAction.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Deployment.WindowsInstaller;
+
+namespace Elastic.PackageCompiler.Beats
+{
+    public class AgentCustomAction
+    {
+        [CustomAction]
+        public static ActionResult InstallAction(Session session)
+        {
+            try
+            {
+                // If there are no install args, we stop here
+                // // (the MSI just copied the files, there will be no agent-install)
+                if (string.IsNullOrEmpty(session["INSTALLARGS"]))
+                {
+                    session.Log("No INSTALLARGS provided, skipping agent install");
+                    return ActionResult.Success;
+                }
+
+                string install_args = session["INSTALLARGS"];
+                string install_folder = Path.Combine(session["INSTALLDIR"], session["exe_folder"]);
+
+                System.Diagnostics.Process process = new System.Diagnostics.Process();
+                process.StartInfo.WorkingDirectory = install_folder;
+                process.StartInfo.FileName = Path.Combine(install_folder, "elastic-agent.exe");
+                process.StartInfo.Arguments = "install -f " + install_args;
```

Comment:
Should we consider always adding --delay-enroll so that installation success doesn't depend on fleet service reachability + valid API token?

Reply:
one of the tasks was to make sure the service was up and running right after the installation flow... Don't you think the admin would want to know if the token is wrong when rolling out agent installations?